A few fixes

mcpicoli opened the pull request:
Fixed the following:

1) Reset procedure changed to reflect the official procedure described in the eWBM's eLR100-UL-00 module documentation. Also, the reset circuitry used by Robocore is different.

2) Solved several "narrowing conversion" warnings. While harmless since there is parameter verification ensuring their validity, those warnings were annoying.

Commit:
- Solved several "narrowing conversion" warnings at compile time.
- Changed the hardware reset procedure to comply with eWBM's documentation.
- Changed the hardware reset procedure to agree with Robocore's hardware (that is different from the reset circuitry suggested by eWBM's documentation)

Commit:
Typo fixed.

gfretin commented:
Hi @mcpicoli , thank you for your suggestions. I have some comments about this pull request.

- Although the **LoRaWAN Bee** has an internal pull-down resistor on its reset pin, leaving the microcontroller's pin as an input isn't always compatible (e.g. ESP32 pin 5), therefore it is safer to make it an output and LOW by default.
- Which IDE and compiler did you use for your work? Did the compiler output a warning for the command `sport[index++] = temp[i] + '0';`?
- The current code has indeed a few narrowing conversions, so it will be nice to correct it with your suggestions.

mcpicoli commented:
Hi,

I could not find any reason as for why GPIO5 should not be left as an input. As per the Espressif's ESP32 [documentation](https://docs.espressif.com/projects/esptool/en/latest/esp32/advanced-topics/boot-mode-selection.html#boot-mode-message), GPIO5 is a strapping pin with an internal pull-up resistor, so it (usually) should be left floating or pulled high at boot. But after boot, there would be no problem at all. You got me curious about that and I'll do a quick test as soon as I get back in the lab...

Additional reference [here](https://esp32.com/viewtopic.php?t=297), since there is no clear explanation about GPIO5 on the previous page.

About the second bullet point, I've used the official Arduino IDE (1.8.19) and the latest (2.0.2) ESP32 SDK.  It was easy to verify that the specific line you mentioned didn't give any warning. Maybe I got a bit overzealous with all similar operations?

P.S.: There are GPIOs that could _never be used_ like VP/VN (GPIO 36/39) since they're inputs only, the other boot strapping pins, and all of the GPIOs usually attached to the flash memory. I believe that unless we go the way of checking (and rejecting at compile time) every invalid pin, it is reasonable to expect that the person designing the circuit should know which GPIOs to avoid. Add to the mix the newer revisions of the ESP32 family and the fact that the library may be used on other architectures and now you have another layer of complications. 

Anyway, while not strictly necessary, the change in the reset procedure was made to comply with the officially recommended method published in the eWBM eRL100-UL00 module's documentation.

mcpicoli commented:
I tested virtually every GPIO on the ESP32's devkit and I found that keeping the RESET pin as an OUTPUT is _more reliable_ than keeping it as an INPUT.

I'll amend the committed code to reflect this.

Commit:
- Undid changes to the reset procedure, as it turns out, it is more reliable to force the pin output to low, as several pins on some architectures have internal pull-up resistors.
- Undid one overzealous attempt to fix "narrowing conversion" warnings that wasn't.

Commit:
Another typo. Another unneeded "narrowing conversion" fix undone.

gfretin commented:
Hi @mcpicoli, thanks again for your work, I've just merged your pull request.
